packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -3183,51 +3184,57 @@ describe("IModelTransformer", () => {
     targetDb.close();
   });
 
-  it("should not update root elements when skipPropagateChangesToRootElements is set to true", async () => {
-    const iModelShared: SnapshotDb =
-      IModelTransformerTestUtils.createSharedIModel(outputDir, ["A", "B"]);
-    const iModelA: SnapshotDb = IModelTransformerTestUtils.createTeamIModel(
-      outputDir,
-      "A",
-      Point3d.create(0, 0, 0),
-      ColorDef.green
-    );
-    IModelTransformerTestUtils.assertTeamIModelContents(iModelA, "A");
-    const iModelExporterA = new IModelExporter(iModelA);
+  for (const skipPropagateChangesToRootElementsTrueByDefault of [true, false]) {
+    it("should not update root elements when skipPropagateChangesToRootElements is set to true", async () => {
+      const iModelShared: SnapshotDb =
+        IModelTransformerTestUtils.createSharedIModel(outputDir, ["A", "B"]);
+      const iModelA: SnapshotDb = IModelTransformerTestUtils.createTeamIModel(
+        outputDir,
+        "A",
+        Point3d.create(0, 0, 0),
+        ColorDef.green
+      );
+      IModelTransformerTestUtils.assertTeamIModelContents(iModelA, "A");
+      const iModelExporterA = new IModelExporter(iModelA);
 
-    const subjectId: Id64String = IModelTransformerTestUtils.querySubjectId(
-      iModelShared,
-      "A"
-    );
-    const transformerA2S = new IModelTransformer(
-      iModelExporterA,
-      iModelShared,
-      {
-        targetScopeElementId: subjectId,
-        danglingReferencesBehavior: "ignore",
-        skipPropagateChangesToRootElements: true,
+      const subjectId: Id64String = IModelTransformerTestUtils.querySubjectId(
+        iModelShared,
+        "A"
+      );
+      let transformerA2S: IModelTransformer;
+      if (skipPropagateChangesToRootElementsTrueByDefault) {
+        transformerA2S = new IModelTransformer(iModelExporterA, iModelShared, {
+          targetScopeElementId: subjectId,
+          danglingReferencesBehavior: "ignore",
+        });
+      } else {
+        transformerA2S = new IModelTransformer(iModelExporterA, iModelShared, {
+          targetScopeElementId: subjectId,
+          danglingReferencesBehavior: "ignore",
+          skipPropagateChangesToRootElements: true,
+        });
       }
-    );
-    transformerA2S.context.remapElement(IModel.rootSubjectId, subjectId);
+      transformerA2S.context.remapElement(IModel.rootSubjectId, subjectId);
 
-    // Act
-    await transformerA2S.processAll();
+      // Act
+      await transformerA2S.processAll();
 
-    // Assert
-    const rootElements = ["0x10", "0xe"];
-    rootElements.forEach((rootElementId) => {
-      const dictionary = iModelShared.elements.getElement(rootElementId);
-      assert.equal(
-        dictionary.parent?.id,
-        "0x1",
-        `Root element '${rootElementId}' parent should not be remapped to '${dictionary.parent?.id}'.`
-      );
-    });
+      // Assert
```

Comment:
Is there a set to false case? Maybe you can loop over [true, false, undefined] instead of [true, false].

If undefined you pass nothing and rely on defaults, otherwise you pass true / false to skipPropagate.

Reply:
resolved in [commit adbbdd68467f8525f12e36087c0b013e68e88862](https://github.com/iTwin/imodel-transformer/pull/188/commits/adbbdd68467f8525f12e36087c0b013e68e88862)